service/matching/pri_task_reader.go

```diff
@@ -518,11 +518,11 @@ func (tr *priTaskReader) doGC() {
 	ctx, cancel := context.WithTimeout(tr.backlogMgr.tqCtx, ioTimeout)
 	defer cancel()
 
-	n, err := tr.backlogMgr.db.CompleteTasksLessThan(ctx, tr.ackLevel+1, batchSize, tr.subqueue)
```

Comment:
Several unit tests produced a data race if `tr.ackLevel` is read outside of the lock.

Reply:
I see, but we can't do io inside tr.lock, that's very deliberate. we should pass in the ack level in from maybeGCLocked

Reply:
Let me try that 👍